### ISSUE TITLE
Crash on iOS 14 and 13

### ISSUE THREAD

**Issue by tqu-dev** [RevenueCat/purchases-ios#2004]
Using SDK 4.13.4 (same bug also with 4.13.2; I have not tested other versions), xCode 14.0.0 or 14.0.1:

Apps using the RevenueCat XCFramework are immediately crashing on app start on iOS 13 and 14 but they work fine on iOS 15 and 16.

Here is the debug output:
```
Oct 27 05:58:38 myapp(libswiftFoundation.dylib)[2768] <Notice>: [Purchases] - DEBUG: \M-b\M^D\M-9\M-o\M-8\M^O Looks like you're using a legacy API key.
Oct 27 05:58:38 myapp(libswiftFoundation.dylib)[2768] <Notice>: This is still supported, but it's recommended to migrate to using platform-specific API key, which should look like 'appl_1a2b3c4d5e6f7h'.
Oct 27 05:58:38 myapp(libswiftFoundation.dylib)[2768] <Notice>: See https://rev.cat/auth for more details.
Oct 27 05:58:38 myapp(libnetwork.dylib)[2768] <Notice>: networkd_settings_read_from_file initialized networkd settings by reading plist directly
Oct 27 05:58:38 myapp(libnetwork.dylib)[2768] <Notice>: networkd_settings_read_from_file initialized networkd settings by reading plist directly
Oct 27 05:58:38 myapp(libswiftFoundation.dylib)[2768] <Notice>: [Purchases] - DEBUG: \M-p\M^_\M^Q\M-$ Identifying App User ID
Oct 27 05:58:38 myapp(libswiftFoundation.dylib)[2768] <Notice>:     HighFrequency = 1;
Oct 27 05:58:38 myapp(libswiftFoundation.dylib)[2768] <Notice>:     bundleID = "com.myappid";
Oct 27 05:58:38 myapp(libswiftFoundation.dylib)[2768] <Notice>:     pid = 2768;
Oct 27 05:58:38 myapp(libswiftFoundation.dylib)[2768] <Notice>: } state:0x1 events:0 mask:0x0
Oct 27 05:58:38 myapp(FrontBoardServices)[2768] <Notice>: [FBSDisplaySource 1-1] silently connecting <FBSDisplayConfiguration: 0x2811f82a0; Main; mode: "428x926@3x 60Hz p3 SDR">
Oct 27 05:58:38 myapp(FrontBoardServices)[2768] <Notice>: [FBSDisplaySource 1-1] initialized <FBSDisplayConfiguration: 0x2811f82a0; Main; mode: "428x926@3x 60Hz p3 SDR">
Oct 27 05:58:38 myapp(XCTTargetBootstrap)[2768] <Notice>: Registering for test daemon availability notify post.
Oct 27 05:58:38 myapp(XCTTargetBootstrap)[2768] <Notice>: notify_get_state check indicated test daemon not ready.
Oct 27 05:58:38 myapp(XCTTargetBootstrap)[2768] <Notice>: 	<RBSLegacyAttribute| requestedReason:FinishTask reason:FinishTask flags:( PreventTaskSuspend )>,
Oct 27 05:58:38 myapp(XCTTargetBootstrap)[2768] <Notice>: 	<RBSAcquisitionCompletionAttribute| policy:AfterValidation>
Oct 27 05:58:38 myapp(XCTTargetBootstrap)[2768] <Notice>: 	]>
Oct 27 05:58:38 myapp(UIKitCore)[2768] <Error>: Unknown class ViewController in Interface Builder file.
Oct 27 05:58:38 myapp(UIKitCore)[2768] <Error>: The app delegate must implement the window property if it wants to use a main storyboard file.
Oct 27 05:58:38 myapp(libswiftFoundation.dylib)[2768] <Notice>: [Purchases] - DEBUG: \M-b\M^D\M-9\M-o\M-8\M^O Looks like you're using a legacy API key.
Oct 27 05:58:38 myapp(libswiftFoundation.dylib)[2768] <Notice>: This is still supported, but it's recommended to migrate to using platform-specific API key, which should look like 'appl_1a2b3c4d5e6f7h'.
Oct 27 05:58:38 myapp(libswiftFoundation.dylib)[2768] <Notice>: See https://rev.cat/auth for more details.
Oct 27 05:58:38 myapp(libnetwork.dylib)[2768] <Notice>: networkd_settings_read_from_file initialized networkd settings by reading plist directly
Oct 27 05:58:38 myapp(libnetwork.dylib)[2768] <Notice>: networkd_settings_read_from_file initialized networkd settings by reading plist directly
Oct 27 05:58:38 myapp(libswiftFoundation.dylib)[2768] <Notice>: [Purchases] - DEBUG: \M-p\M^_\M^Q\M-$ Identifying App User ID
Oct 27 05:58:38 myapp(libswiftFoundation.dylib)[2768] <Notice>:     HighFrequency = 1;
Oct 27 05:58:38 myapp(libswiftFoundation.dylib)[2768] <Notice>:     bundleID = "com.myappid";
Oct 27 05:58:38 myapp(libswiftFoundation.dylib)[2768] <Notice>:     pid = 2768;
Oct 27 05:58:38 myapp(libswiftFoundation.dylib)[2768] <Notice>: } state:0x1 events:0 mask:0x0
Oct 27 05:58:38 ReportCrash[2767] <Notice>: Formulating fatal report for corpse[2768] myapp

```

**Comment by RCGitBot** [comment]
👀 [SDKONCALL-158] We've just linked this issue to our internal tracker and notified the team. Thank you for reporting, we're checking this out!

**Comment by Code43210** [comment]
Sorry, I do not have an iOS 13/14 device anymore but used a remote testing service with TestFlight which returns only this. If you want then I can send you a TestFlight invitation via private message.

**Comment by Code43210** [comment]
The problem happens with all my apps. They use objective-c:

```
+ (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
{
    RCPurchases.logLevel = RCLogLevelError;
    RCConfigurationBuilder *configuration = [RCConfiguration builderWithAPIKey:[NSString stringWithCString:REV_CAT_API_KEY encoding:NSASCIIStringEncoding]];
    //configuration = [configuration withAppUserID:@"<app_user_id>"];
    [RCPurchases configureWithConfiguration:[configuration build]];
}
```

**Comment by Code43210** [comment]
Do you need further information?

**Comment by Code43210** [comment]
The problem did not happen with the old 3.x SDKs.

**Comment by NachoSoto** [comment]
We'll need a stack trace from the crash in order to investigate, the console log alone here has no information.
Are you no longer able to reproduce?

**Comment by Code43210** [comment]
Here is the output using an iOS simulator:
```
dyld: lazy symbol binding failed: can't resolve symbol _$sScMMa in /Users/ham/Desktop/Development/Mac/build-goprobluetooth-Qt_6_4_0_for_iOS_Simulator/Debug-iphonesimulator/RevenueCat.framework/RevenueCat because dependent dylib @rpath/libswift_Concurrency.dylib could not be loaded
dyld: can't resolve symbol _$sScMMa in /Users/ham/Desktop/Development/Mac/build-goprobluetooth-Qt_6_4_0_for_iOS_Simulator/Debug-iphonesimulator/RevenueCat.framework/RevenueCat because dependent dylib @rpath/libswift_Concurrency.dylib could not be loaded
CoreSimulator 857.7 - Device: iOS 14 (E4F8877E-2E35-460C-B9E5-BA5C7C93C0E5) - Runtime: iOS 14.5 (18E182) - DeviceType: iPhone 6s
```

"Always Embed Swift Standard Libraries" is set to "YES" in xCode.

and the stack:
```
Thread 1 Queue : com.apple.main-thread (serial)
#0	0x0000000120d9283a in __abort_with_payload ()
#1	0x0000000120dab9c7 in abort_with_payload_wrapper_internal ()
#2	0x0000000120dab9f9 in abort_with_payload ()
#3	0x000000011ad2bef8 in abort_with_payload ()
#4	0x000000011ace26bf in dyld::halt(char const*) ()
#5	0x000000011ace2849 in dyld::fastBindLazySymbol(ImageLoader**, unsigned long) ()
#6	0x00007fff2025eed9 in _dyld_fast_stub_entry(void*, long) ()
#7	0x00007fff2025d5a6 in dyld_stub_binder ()
#8	0x000000011b154000 in 0x11b154000 ()
#9	0x000000011b08108b in specialized PurchasesOrchestrator.init(productsManager:paymentQueueWrapper:systemInfo:subscriberAttributes:operationDispatcher:receiptFetcher:customerInfoManager:backend:currentUserProvider:transactionsManager:deviceCache:offeringsManager:manageSubscriptionsHelper:beginRefundRequestHelper:) [inlined] ()
#10	0x000000011b081056 in PurchasesOrchestrator.init(productsManager:paymentQueueWrapper:systemInfo:subscriberAttributes:operationDispatcher:receiptFetcher:customerInfoManager:backend:currentUserProvider:transactionsManager:deviceCache:offeringsManager:manageSubscriptionsHelper:beginRefundRequestHelper:) [inlined] at /Users/distiller/purchases-ios/<compiler-generated>:134
#11	0x000000011b081056 in specialized PurchasesOrchestrator.__allocating_init(productsManager:paymentQueueWrapper:systemInfo:subscriberAttributes:operationDispatcher:receiptFetcher:customerInfoManager:backend:currentUserProvider:transactionsManager:deviceCache:offeringsManager:manageSubscriptionsHelper:beginRefundRequestHelper:) [inlined] at /Users/distiller/purchases-ios/Sources/Purchasing/Purchases/PurchasesOrchestrator.swift:134
#12	0x000000011b081056 in PurchasesOrchestrator.__allocating_init(productsManager:paymentQueueWrapper:systemInfo:subscriberAttributes:operationDispatcher:receiptFetcher:customerInfoManager:backend:currentUserProvider:transactionsManager:deviceCache:offeringsManager:manageSubscriptionsHelper:beginRefundRequestHelper:) [inlined] at /Users/distiller/purchases-ios/<compiler-generated>:134
#13	0x000000011b081056 in closure #1 in Purchases.init(apiKey:appUserID:userDefaults:observerMode:platformInfo:storeKit2Setting:storeKitTimeout:networkTimeout:dangerousSettings:) at /Users/distiller/purchases-ios/Sources/Purchasing/Purchases/Purchases.swift:343
#14	0x000000011b08eb73 in specialized Purchases.__allocating_init(apiKey:appUserID:userDefaults:observerMode:platformInfo:storeKit2Setting:storeKitTimeout:networkTimeout:dangerousSettings:) at /Users/distiller/purchases-ios/Sources/Purchasing/Purchases/Purchases.swift:322
#15	0x000000011b090dfd in Purchases.__allocating_init(apiKey:appUserID:userDefaults:observerMode:platformInfo:storeKit2Setting:storeKitTimeout:networkTimeout:dangerousSettings:) [inlined] ()
#16	0x000000011b090dcd in specialized static Purchases.configure(withAPIKey:appUserID:observerMode:userDefaults:platformInfo:storeKit2Setting:storeKitTimeout:networkTimeout:dangerousSettings:) [inlined] at /Users/distiller/purchases-ios/Sources/Purchasing/Purchases/Purchases.swift:952
#17	0x000000011b090dcd in static Purchases.configure(withAPIKey:appUserID:observerMode:userDefaults:platformInfo:storeKit2Setting:storeKitTimeout:networkTimeout:dangerousSettings:) [inlined] ()
#18	0x000000011b090dcd in specialized static Purchases.configure(with:) at /Users/distiller/purchases-ios/Sources/Purchasing/Purchases/Purchases.swift:883
#19	0x000000011b08894e in static Purchases.configure(with:) [inlined] ()
#20	0x000000011b088946 in @objc static Purchases.configure(with:) at /Users/distiller/purchases-ios/<compiler-generated>:882
#21	0x000000010d829b84 in IOSStoreHandler::initFramework() at /Users/ham/Desktop/Development/Mac/goprobluetooth/appbasis/licensing/iosstorehandler.mm:86
#22	0x000000010d6edabf in +[QIOSBaseApplicationTools application:didFinishLaunchingWithOptions:] at /Users/ham/Desktop/Development/Mac/goprobluetooth/appbasis/core/ios/qiosbaseapplicationtools.mm:65
#23	0x000000010d6ed6a4 in -[QIOSApplicationDelegate(VMDelegate) application:didFinishLaunchingWithOptions:] at /Users/ham/Desktop/Development/Mac/goprobluetooth/appbasis/core/ios/qiosapplicationdelegate+vmdelegate.mm:19
#24	0x00007fff246bd6f1 in -[UIApplication _handleDelegateCallbacksWithOptions:isSuspended:restoreState:] ()
#25	0x00007fff246bf33b in -[UIApplication _callInitializationDelegatesWithActions:forCanvas:payload:fromOriginatingProcess:] ()
#26	0x00007fff246c4d63 in -[UIApplication _runWithMainScene:transitionContext:completion:] ()
#27	0x00007fff23ce629b in -[_UISceneLifecycleMultiplexer completeApplicationLaunchWithFBSScene:transitionContext:] ()
#28	0x00007fff24272726 in _UIScenePerformActionsWithLifecycleActionMask ()
#29	0x00007fff23ce6dad in __101-[_UISceneLifecycleMultiplexer _evalTransitionToSettings:fromSettings:forceExit:withTransitionStore:]_block_invoke ()
#30	0x00007fff23ce686e in -[_UISceneLifecycleMultiplexer _performBlock:withApplicationOfDeactivationReasons:fromReasons:] ()
#31	0x00007fff23ce6bda in -[_UISceneLifecycleMultiplexer _evalTransitionToSettings:fromSettings:forceExit:withTransitionStore:] ()
#32	0x00007fff23ce646f in -[_UISceneLifecycleMultiplexer uiScene:transitionedFromState:withTransitionContext:] ()
#33	0x00007fff23cee6d8 in __186-[_UIWindowSceneFBSSceneTransitionContextDrivenLifecycleSettingsDiffAction _performActionsForUIScene:withUpdatedFBSScene:settingsDiff:fromSettings:transitionContext:lifecycleActionType:]_block_invoke ()
#34	0x00007fff2417c77d in +[BSAnimationSettings(UIKit) tryAnimatingWithSettings:actions:completion:] ()
#35	0x00007fff2428f60e in _UISceneSettingsDiffActionPerformChangesWithTransitionContext ()
#36	0x00007fff23cee3de in -[_UIWindowSceneFBSSceneTransitionContextDrivenLifecycleSettingsDiffAction _performActionsForUIScene:withUpdatedFBSScene:settingsDiff:fromSettings:transitionContext:lifecycleActionType:] ()
#37	0x00007fff23b117d5 in __64-[UIScene scene:didUpdateWithDiff:transitionContext:completion:]_block_invoke ()
#38	0x00007fff23b1024e in -[UIScene _emitSceneSettingsUpdateResponseForCompletion:afterSceneUpdateWork:] ()
#39	0x00007fff23b1140e in -[UIScene scene:didUpdateWithDiff:transitionContext:completion:] ()
#40	0x00007fff246c3203 in -[UIApplication workspace:didCreateScene:withTransitionContext:completion:] ()
#41	0x00007fff241a4ab4 in -[UIApplicationSceneClientAgent scene:didInitializeWithEvent:completion:] ()
#42	0x00007fff25a1b40b in -[FBSScene _callOutQueue_agent_didCreateWithTransitionContext:completion:] ()
#43	0x00007fff25a43e55 in __94-[FBSWorkspaceScenesClient createWithSceneID:groupID:parameters:transitionContext:completion:]_block_invoke.176 ()
#44	0x00007fff25a28f12 in -[FBSWorkspace _calloutQueue_executeCalloutFromSource:withBlock:] ()
#45	0x00007fff25a43b28 in __94-[FBSWorkspaceScenesClient createWithSceneID:groupID:parameters:transitionContext:completion:]_block_invoke ()
#46	0x000000011b1d074e in _dispatch_client_callout ()
#47	0x000000011b1d3656 in _dispatch_block_invoke_direct ()
#48	0x00007fff25a695d0 in __FBSSERIALQUEUE_IS_CALLING_OUT_TO_A_BLOCK__ ()
#49	0x00007fff25a692b6 in -[FBSSerialQueue _targetQueue_performNextIfPossible] ()
#50	0x00007fff25a6977b in -[FBSSerialQueue _performNextFromRunLoopSource] ()
#51	0x00007fff20390ede in __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ ()
#52	0x00007fff20390dd6 in __CFRunLoopDoSource0 ()
#53	0x00007fff2039029e in __CFRunLoopDoSources0 ()
#54	0x00007fff2038a9f7 in __CFRunLoopRun ()
#55	0x00007fff2038a1a7 in CFRunLoopRunSpecific ()
#56	0x00007fff2b874d85 in GSEventRunModal ()
#57	0x00007fff246c14df in -[UIApplication _run] ()
#58	0x00007fff246c639c in UIApplicationMain ()
#59	0x000000010ed8b6a4 in qt_main_wrapper at /Users/qt/work/qt/qtbase/src/plugins/platforms/ios/qioseventdispatcher.mm:206
#60	0x0000000120d2f310 in start ()
```

**Comment by Code43210** [comment]
Setting "Always Embed Swift Standard Libraries" to "YES" and manually including libswift_Concurrency.tbd fixes the problem.

**Comment by aboedo** [comment]
Another approach for this is to add a single, empty Swift file to your project. That enables Swift support as well. 
It's a bummer that the error message when this happens is so unclear. 

@Code43210 what Unity version are you using? We've had our approaches to automating this part so developers never run into it, but it seems like results vary by Unity version. This [post install script](https://github.com/RevenueCat/purchases-unity/blob/main/RevenueCat/Editor/RevenueCatPostInstall.cs) is supposed to take care of that. Could you confirm whether that file was added to your project, as well as the Unity version?

**Comment by Code43210** [comment]
I am not using Unity but Qt (https://www.qt.io/). Adding a single Swift file did not work for me. Instead I have to do these steps:

1. Drag the RevenueCat.xcframeworks into xCode.
2. Set ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES to YES.
3. Add the libswift_Concurrency.tbd as an optional library.

**Comment by roostr** [comment]
I also faced this issue, and the solutions listed above did not help me.

I would see a crash, and the console would say:

```
… because dependent dylib @rpath/libswift_Concurrency.dylib could not be loaded
```

Here’s how I fixed it:

1. Project Settings
2. Select your target
3. Build Settings
4. Linking
5. Runpath Search Paths
6. Set value to:

```
$(inherited)
@executable_path/Frameworks
```

<img width="751" alt="Screenshot 2022-12-07 at 10 45 09 AM" src="https://user-images.githubusercontent.com/978146/206224767-e5d4b0e6-5ecc-4c99-8192-92978324563b.png">
